#! /usr/bin/env python

PACKAGE = 'hamal_diff_drive_controller'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t

gen = ParameterGenerator()

gen.add(
    "max_vel_x",
    double_t,
    0,
    "Maximum linear velocity in the X axis.",
    1.0
)

gen.add(
    "min_vel_x",
    double_t,
    0,
    "Minimum linear velocity in the X axis.",
    1.0
)

gen.add(
    "max_accel_x",
    double_t,
    0,
    "Maximum linear acceleration in the X axis.",
    1.0
)

gen.add(
    "min_accel_x",
    double_t,
    0,
    "Minimum linear acceleration in the X axis.",
    1.0
)

gen.add(
    "max_jerk_x",
    double_t,
    0,
    "Maximum linear jerk in the X axis.",
    1.0
)

gen.add(
    "min_jerk_x",
    double_t,
    0,
    "Minimum linear jerk in the X axis.",
    1.0
)

gen.add(
    "max_vel_z",
    double_t,
    0,
    "Maximum angular velocity in the Z axis.",
    1.0
)

gen.add(
    "min_vel_z",
    double_t,
    0,
    "Minimum angular velocity in the Z axis.",
    1.0
)

gen.add(
    "max_accel_z",
    double_t,
    0,
    "Maximum angular acceleration in the Z axis.",
    1.0
)

gen.add(
    "min_accel_z",
    double_t,
    0,
    "Minimum angular acceleration in the Z axis.",
    1.0
)

gen.add(
    "max_jerk_z",
    double_t,
    0,
    "Maximum angular jerk in the Z axis.",
    1.0
)

gen.add(
    "min_jerk_z",
    double_t,
    0,
    "Minimum angular jerk in the Z axis.",
    1.0
)

exit(gen.generate(PACKAGE, "hamal_diff_drive_controller", "HamalDiffDriveController"))